src/plugin.js

```diff
@@ -73,7 +74,12 @@ export default {
     } else {
       $ld = initialize({ clientSideId, user, ldOptions, readyBeforeIdentify });
     }
-    // eslint-disable-next-line no-param-reassign
-    Vue.prototype.$ld = Vue.observable($ld);
+
+    if (isVue2) {
+      // eslint-disable-next-line no-param-reassign
+      vue.prototype.$ld = $ld;
+    } else {
+      vue.provide('$ld', $ld);
```

Comment:
Why chose to use provide over assigning it through `app.config.globalProperties`?

https://v3.vuejs.org/guide/plugins.html#writing-a-plugin

Reply:
I was having trouble getting components to react to changes with a reactive globalProperty, but revisiting it now it looks like it does work as well so I'll definitely switch back to this.